src/xtgeo/grid3d/grid_properties.py

```diff
@@ -295,21 +508,13 @@ def get_prop_by_name(self, name, raiseserror=True):
 
     def append_props(self, proplist):
         """Add a list of GridProperty objects to current GridProperties instance."""
-        for prop in proplist:
-            if isinstance(prop, GridProperty):
-                # an prop instance can only occur once
-                if prop not in self._props:
-
-                    self._props.append(prop)
-                    self._names.append(prop.name)
-                    self._dates.append(prop.date)
-                    self._ncol = prop.ncol
-                    self._nrow = prop.nrow
-                    self._nlay = prop.nlay
-                else:
-                    logger.info("Not added. GridProperty instance already present")
-            else:
-                raise ValueError("Input property is not a valid GridProperty " "object")
+        if not self._props and proplist:
+            self._ncol = proplist[0].ncol
+            self._nrow = proplist[0].nrow
+            self._nlay = proplist[0].nlay
+        self._props += proplist
```

Comment:
Possible to replace next 3 lines by using ` props` and not `_props` ?

Reply:
Indeed, fixed now :)

Reply:
Attempted this, but the `+=` calls `__iadd__` of  the list and `self.props` will return None if empty so the logic becomes messy.

Reply:
This can be fixed once we have had a breaking change which gives a better getter/setter of props.